src/components/Calendar/helpers/buildNewRangeFromValue.js

```diff
@@ -0,0 +1,27 @@
+import isDateWithinRange from './isDateWithinRange';
+import compareDates from './compareDates';
+
+let rangeIndex = 0;
```

Comment:
we should avoid this kind of side effects and instead embrace functional programming best practice